autogluon/utils/tabular/ml/models/lr/lr_model.py

```diff
@@ -0,0 +1,206 @@
+            self.pipeline.fit(X)
+
+        logger.log(15, " - transforming inputs using pre-processing pipeline")
+        X = self.pipeline.transform(X)
+        logger.log(15, " - output shape %s" % str(X.shape))
+
+        return X
+
+    def _set_default_params(self):
+        default_params = {
+            'model_type': 'LR',
+            'C': 1,
+            'vectorizer_dict_size': 75000,
+            'proc.ngram_range': (1, 5),
+            'proc.skew_threshold': 0.99,
+            'proc.impute_strategy': 'median',  # strategy argument of sklearn.SimpleImputer() used to impute missing numeric values
+        }
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _get_default_searchspace(self, problem_type):
+        spaces = {
+            # 'C': Real(lower=1e-4, upper=1e5, default=1),
+            # 'tokenizer': Categorical('split', 'sentencepiece')
+        }
+        return spaces
+
+    def fit(self, X_train, Y_train, X_test=None, Y_test=None, time_limit=None, **kwargs):
+        hyperparams = self.params.copy()
+
+        self.cat_one_hot = get_one_hot_features(X_train)
+
+        # See solver options here: https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html
+        if self.problem_type == BINARY:
+            Y_train = Y_train.astype(int).values
+            solver = 'lbfgs'  # TODO use liblinear for smaller datasets
+        elif self.problem_type == MULTICLASS:
+            solver = 'saga'  # another option is lbfgs
+        else:
+            solver = 'lbfgs'
+
+        X_train = self.preprocess(X_train, is_train=True, vect_max_features=hyperparams['vectorizer_dict_size'])
+
+        # TODO: get seed from seeds provider
+        self.model = LogisticRegression(C=hyperparams['C'], random_state=17, solver=solver, n_jobs=-1)
+        self.model.fit(X_train, Y_train)
+
+    def hyperparameter_tune(self, X_train, X_test, Y_train, Y_test, scheduler_options=None, **kwargs):
+        self.fit(X_train=X_train, X_test=X_test, Y_train=Y_train, Y_test=Y_test, **kwargs)
+        hpo_model_performances = {self.name: self.score(X_test, Y_test)}
+        hpo_results = {}
+        self.save()
+        hpo_models = {self.name: self.path}
+
+        return hpo_models, hpo_model_performances, hpo_results
+
+
+class LRModelAll(LRModel):
+    def _select_features(self, df, types_of_features, categorical_featnames, language_featnames, continuous_featnames):
+        # continuous = numeric features to rescale
+        # skewed = features to which we will apply power (ie. log / box-cox) transform before normalization
+        # onehot = features to one-hot encode (unknown categories for these features encountered at test-time are encoded as all zeros). We one-hot encode any features encountered that only have two unique values.
+        one_hot_threshold = 2 if len(language_featnames) > 0 else 10000
```

Comment:
Why? Won't this affect categorical non-language features?

Reply:
It was creating memory issues when joined with TFIDF vectors from NLP features. Basically currently it's switching between 2 modes: NLP and Plain. Not the ideal solution, needs deep dive to find proper resolution.